### ISSUE TITLE
Literal dicts aren't joined to a TypedDict the same way as the same TypedDict's constructor

### ISSUE THREAD

**Issue by Gobot1234** [microsoft/pyright#3809]
**Describe the bug**
```py
x: TypedDictClass = {...}
x = TypedDictClass(...)
```
are not treated the same as each other when unpacking from multiple dictionaries.

**To Reproduce**
```py
class Description(TypedDict):
	foo: int

class Asset(TypedDict):
	bar: str

class Item(Description, Asset):
	...

description: Description = {"foo": 12345}
asset: Asset = {"bar": "baz"}

data: Item = {**description, **asset}  # not fine # Expression of type "dict[str, object]" cannot be assigned to declared type "Item"
data = Item(**description, **asset)  # fine
```

**Expected behavior**
Both should be treated the same way

**VS Code extension or command-line**
Pylance v2022.8.10

**Comment by erictraut** [comment]
From a type checking perspective, these two cases are very different. One involves bidirectional type inference of a dictionary expression. The other involves argument matching for keyword parameters in a constructor. 

Adding support for bidirectional type inference of a dictionary expression with unpacked TypedDict instances is a reasonable feature request, but I don't know how common such usage would be. Mypy doesn't support it currently. 

I'll look into it and see if I can add this functionality without adding significant complexity.

**Comment by erictraut** [comment]
This will be included in the next release.

**Comment by erictraut** [comment]
This is fixed in pyright 1.1.267, which I just published. It will also be included in a future version of pylance.
